### PR TITLE
Add initial code path and test for overflow page redesign

### PR DESCRIPTION
**Pull request by nanotone**
## Summary

*This work is behind a feature toggle (flipper): NO* -- this work is not hooked up to live code paths; when it is, it will be behind a feature toggle.

This PR adds the ability to generate a redesigned PDF form overflow (a.k.a. "extras") page, without affecting the existing overflow page generator used by all of VA.gov's existing PDFs. The new design is enabled by passing `extras_redesign: true` to PdfFill. For now, **only a minor formatting change has been implemented, and is visible only by running the PdfFill spec**.

The new design is intended to be rolled out carefully on a form-by-form basis, starting with the 21-0781V2 which is going live later this year. I am on the Employee Experience team. The 0781-specific parts are owned by DBEX 2 (Carbs) and the remainder is owned by Platform.

## Related issue(s)

- department-of-veterans-affairs/abd-vro#4013

## Testing done

- [x] *New code is covered by unit tests*
- Prior behavior: only the old overflow page generator can be used.
- New behavior: the new overflow page generator is used by passing `extras_redesign: true` to PdfFill -- currently done only in tests.
- To manually inspect the overflow page produced by the new generator, run `filler_spec.rb` and prevent or interrupt the PDF cleanup either by comment out `File.delete(extras_path)` or by tampering with the `overflow_redesign_extras.pdf` fixture. The PDF generated by the spec will be left in `tmp/pdfs/`.

## Screenshots
"Before and after" files will always be checked into Git as `overflow_extras.pdf` and `overflow_redesign_extras.pdf` respectively. For convenience, here are some screenshots.

Before:
Screenshot 2025-02-05 at 13 18 09

After:
Screenshot 2025-02-05 at 13 15 15

## What areas of the site does it impact?
* No user-facing impact.

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  I added a screenshot of the developed feature

### CODE REVIEW

File: spec/lib/pdf_fill/filler_spec.rb [hunk 1]
```diff
@@ -73,7 +73,9 @@
   describe '#fill_ancillary_form', run_at: '2017-07-25 00:00:00 -0400' do
     %w[21-4142 21-0781a 21-0781 21-0781V2 21-8940 28-8832 28-1900 21-674 21-0538 26-1880 5655].each do |form_id|
       context "form #{form_id}" do
-        %w[simple kitchen_sink overflow].each do |type|
+        form_types = %w[simple kitchen_sink overflow].product([false])
+        form_types << ['overflow', true] if form_id == '21-0781V2'
+        form_types.each do |type, extras_redesign|
           context "with #{type} test data" do
```
Comment (dfitchett) [inline spec/lib/pdf_fill/filler_spec.rb:79]: Can we modify this to show in the context description if the extras_redesign is set to true? 
`          context "with #{type} test data with extras_redesign #{extras_redesign}" do
`

**Review by dfitchett — APPROVED** [review]
LGTM! Just one suggestion!

**Review by dfitchett — APPROVED** [review]

**Review by rmtolmach — APPROVED** [review]
